assets/admin/students/student-modal/course-list.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * WordPress dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+import { CheckboxControl, Spinner } from '@wordpress/components';
+import { useEffect, useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Course list.
+ */
+export const CourseList = () => {
+	const [ isFetching, setIsFetching ] = useState( true );
+	const [ courses, setCourses ] = useState( [] );
+
+	// Fetch the courses.
+	useEffect( () => {
+		setIsFetching( true );
+
+		apiFetch( {
+			path: '/wp/v2/courses?per_page=100',
+			method: 'GET',
+		} )
+			.then( ( result ) => {
+				setCourses( result );
+			} )
+			.finally( () => {
+				setIsFetching( false );
+			} );
+	}, [] );
+
+	if ( isFetching ) {
+		return (
+			<div className="sensei-student-modal__course-list--loading">
+				<Spinner />
+			</div>
+		);
+	}
+
+	if ( 0 === courses.length ) {
+		return <p>{ __( 'No courses found.', 'sensei-lms' ) }</p>;
+	}
+
+	const coursesMap = ( course ) => {
```

Comment:
Cometic suggestions:
- Rename course map to `Course` or `CourseOption`.
- Move it to the outside of the CourseList to avoid recreating the instance every time the component is rendered.

Reply:
I suggested the initial cap because it is a CourseItem(singular) component and usually the component name is singular.

But no big deal with it.

Reply:
Just to clarify, my suggestion is lowercase + the `to` prefix, if we want to represent a transformation

 `courses.map(toCourseItem)`
 
Or follow the component naming with uppercase.
` courses.map(CourseItem)`

But zero impact with the final results